### PR TITLE
User defined results path

### PR DESCRIPTION
**Pull request by lucamaraschi**
Added `dataPath` option to support user defined path where to save the collect data. The default value was `.`(current folder).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -51,25 +53,32 @@ class ClinicDoctor extends events.EventEmitter {
...
+      env: Object.assign({}, process.env, customEnv)
     })
 
     if (this.detectPort) {
       proc.stdio[3].once('data', data => this.emit('port', Number(data), proc, () => proc.stdio[3].destroy()))
     }
 
     // get logging directory structure
-    const paths = getLoggingPaths({ identifier: proc.pid })
+    const options = this.dataPath ? { identifier: proc.pid, path: this.dataPath } : { identifier: proc.pid }
```
Comment (mcollina) [inline index.js]: this ternary should not be needed

File: injects/sampler.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 
 // create encoding files and directory
-const paths = getLoggingPaths({ identifier: process.pid })
-fs.mkdirSync(paths['/'])
+let paths = null
+if (process.env.NODE_CLINIC_DOCTOR_DATA_PATH) {
+  paths = getLoggingPaths({ path: process.env.NODE_CLINIC_DOCTOR_DATA_PATH, identifier: process.pid })
+} else {
+  paths = getLoggingPaths({ identifier: process.pid })
+}
```
Comment (mcollina) [inline injects/sampler.js]: this if should not be needed, we should be able to pass `path: process.env.NODE_CLINIC_DOCTOR_DATA_PATH` without an if.

File: collect/get-logging-paths.js [hunk 3]
```diff
@@ -5,7 +5,11 @@ const path = require('path')
 function getLoggingPaths (options) {
   let dirpath, basename
   if (options.hasOwnProperty('identifier')) {
-    dirpath = ''
+    if (options.hasOwnProperty('path') && options.path) {
+      dirpath = options.path
+    } else {
+      dirpath = './'
```
Comment (mcollina) [inline collect/get-logging-paths.js]: this should not be needed and I would prefer it to stay as `''`.

Comment (lucamaraschi) [reply collect/get-logging-paths.js]: I think having an explicit path makes the code more readable. Why would you use `''` instead of `'./'`?

Comment (mcollina) [reply collect/get-logging-paths.js]: I am afraid of Windows.

Comment (mcollina) [reply collect/get-logging-paths.js]: Anyway, CI is passing, so +1.

File: README.md [hunk 4]
```diff
@@ -48,6 +48,7 @@ const doctor = new ClinicDoctor()
   * sampleInterval [`<number>`][] Time between each sample in milliseconds.
     **Default**: 10
   * detectPort [`<boolean>`][] **Default**: false
+  * dataPath [`<string>`][] The folder where the collected data is stored. **Default**: '.'
```
Comment (BridgeAR) [inline README.md]: Suggestion: rename the option to `path` as there are no other `path` folders I can think off. Internally it's also called `path`.

File: injects/sampler.js [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
+const makeDir = require('mkdirp')
 const systemInfo = require('../collect/system-info.js')
 const ProcessStat = require('../collect/process-stat.js')
 const getLoggingPaths = require('../collect/get-logging-paths.js')
 const ProcessStatEncoder = require('../format/process-stat-encoder.js')
 
 // create encoding files and directory
-const paths = getLoggingPaths({ identifier: process.pid })
-fs.mkdirSync(paths['/'])
+let paths = getLoggingPaths({ path: process.env.NODE_CLINIC_DOCTOR_DATA_PATH, identifier: process.pid })
```
Comment (BridgeAR) [inline injects/sampler.js]: This looks like it should still be const?

File: collect/get-logging-paths.js [hunk 6]
```diff
@@ -4,10 +4,14 @@ const path = require('path')
 
 function getLoggingPaths (options) {
   let dirpath, basename
-  if (options.hasOwnProperty('identifier')) {
-    dirpath = ''
+  if (options.hasOwnProperty('identifier') && options.identifier) {
```
Comment (BridgeAR) [inline collect/get-logging-paths.js]: :+1:

**Pull request merged**